unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarsLODController.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL
+{
+    public class AvatarsLODController : MonoBehaviour
+    {
+        public static AvatarsLODController i { get; private set; }
+
+        private const float LODS_LOCAL_Y_POS = 1.8f;
+        private const float LODS_VERTICAL_MOVEMENT = 0.1f;
+        private const float LODS_VERTICAL_MOVEMENT_DELAY = 1f;
+
+        public float lodDistance = 16f;
+        public int maxNonLODAvatars = 20;
+
+        private List<AvatarShape> avatarsList = new List<AvatarShape>();
+        private float lastLODsVerticalMovementTime = -1;
+
+        void Awake()
+        {
+            if (i != null)
+            {
+                Destroy(gameObject);
+                return;
+            }
+            i = this;
+
+            enabled = false;
+            KernelConfig.i.EnsureConfigInitialized()
+                        .Then(config =>
+                        {
+                            // We need this check because in the CI the KernelConfig tests may have this object destroyed somehow
+                            if (i == null || this == null)
+                                return;
+
+                            if (config.features.enableAvatarLODs)
+                            {
+                                enabled = true;
+                                CommonScriptableObjects.playerUnityPosition.OnChange += OnMainPlayerReposition;
+                            }
+                            else
+                            {
+                                enabled = false;
+                            }
+                        });
+        }
```

Comment:
Done, check it out, although I'm not so sure everyone will like this new "responsibility" inside the KernelConfig.

Reply:
It's not a new responsibility, `KernelConfig` was holding this data previosuly; now only bypasses it. This allow us to avoid communicating systems when they only want to transfer data

Reply:
In the end I had to revert the change because with master's latest assemblies refactor there is a magical cyclic reference when referencing the DataStore assembly in the KernelConfig assembly, even though the DataStore doesn't use the KernelConfig at all  `¯\_(ツ)_/¯`

Reply:
This should be impossible, `DataStore` should be a "leaf" assembly only referencing models and therefore avoiding cycles. Let's not block this PR but create a ticket to solve the issue.